Source/Validation.swift

```diff
@@ -72,22 +72,8 @@ extension Request {
     @discardableResult
     public func validate<S: Sequence>(statusCode acceptableStatusCode: S) -> Self where S.Iterator.Element == Int {
         return validate { _, response in
-            if acceptableStatusCode.contains(response.statusCode) {
-                return .success
-            } else {
-                let failureReason = "Response status code was unacceptable: \(response.statusCode)"
-
-                let error = NSError(
-                    domain: ErrorDomain,
-                    code: ErrorCode.statusCodeValidationFailed.rawValue,
-                    userInfo: [
-                        NSLocalizedFailureReasonErrorKey: failureReason,
-                        ErrorUserInfoKeys.StatusCode: response.statusCode
-                    ]
-                )
-
-                return .failure(error)
-            }
+            return acceptableStatusCode.contains(response.statusCode) ?
+                .success : .failure(AFError.responseValidationFailed(reason: .unacceptableStatusCode(code: response.statusCode)))
```

Comment:
This is long enough I'd probably just use an `if` / `else` condition rather than a multi-line ternary. Thoughts?